examples/custom_client_trace.rs

```diff
@@ -0,0 +1,115 @@
+// Custom client example with TraceLayer.
+use std::time::Duration;
+
+use futures::{StreamExt, TryStreamExt};
+use http::{Request, Response};
+use hyper::{client::HttpConnector, Body};
+use hyper_tls::HttpsConnector;
+use k8s_openapi::api::core::v1::Pod;
+use serde_json::json;
+use tower::ServiceBuilder;
+use tower_http::{decompression::DecompressionLayer, trace::TraceLayer};
+use tracing::Span;
+
+use kube::{
+    api::{Api, DeleteParams, ListParams, PostParams, ResourceExt, WatchEvent},
+    service::SetBaseUriLayer,
+    Client, Config,
+};
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    std::env::set_var("RUST_LOG", "info,kube=debug,custom_client_trace=debug");
+    tracing_subscriber::fmt::init();
+
+    let config = Config::infer().await?;
+    // Create HttpsConnector using `native_tls::TlsConnector` based on `Config`.
+    let https = {
+        let tls = tokio_native_tls::TlsConnector::from(config.native_tls_connector()?);
+        let mut http = HttpConnector::new();
+        http.enforce_http(false);
+        HttpsConnector::from((http, tls))
+    };
+    let client = Client::new(
+        ServiceBuilder::new()
+            .layer(SetBaseUriLayer::new(config.cluster_url))
+            // Add `DecompressionLayer` to make request headers interesting.
+            .layer(DecompressionLayer::new())
+            .layer(
+                TraceLayer::new_for_http()
+                    .make_span_with(|request: &Request<Body>| {
+                        tracing::debug_span!(
```

Comment:
maybe link the semantic convention doc here: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-client